posts/2024-xx-xx-spring-boot-3.adoc

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: "Running a Spring Boot 3.x application WAR file on Liberty"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/scottkurz
+author_github: https://github.com/scottkurz
+seo-title: Running a Spring Boot 3.x application WAR file on Liberty
+seo-description: TODO In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post use the Spring Boot Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty. 
+blog_description: TODO In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post uses the Spring Boot Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty. 
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Running a Spring Boot 3.x application WAR file on Liberty
+Scott Kurz <https://github.com/scottkurz>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+As of Liberty version 23.0.0.9 there are now two ways to deploy Spring Boot WAR files on Liberty.
+
+While a Spring Boot WAR can be deployed like any standard Jakarta EE WAR, it can also be deployed using an optimized deployment for Spring Boot applications. This deployment adds some additional configuration options, and provides advantages for efficiently building Docker container images.
+
+In this blog post we will step through a simple flow starting from the Spring Initializr.  We will deploy this to Liberty via a standard WAR deployment without any Liberty modifications to the app.  
+
+From there we'll show how to deploy using the optimized Spring Boot deployment by adding the liberty-maven-plugin and some minimal config to the pom.xml, and simple server.xml configuration using the `springBoot-3.0` feature.   We will then be able to build an efficiently-layered Docker image.
+
+Much of the Liberty documentation relating to Spring Boot focuses on the optimized deployment associated with the Spring Boot features, and so we spend some extra time in this post explaining both options.
```

Comment:
Can you add a link to that Liberty doc?

Reply:
And also say why the Liberty doc focuses mainly on the optimized approach and, conversely, why it's worth covering both in this post? Otherwise it sounds like you're criticising the Liberty doc (instead of just fixing the doc), and it's not clear why the Liberty doc would do one thing and you'd do something else in the blog post?